### PR TITLE
Sample 05 - Linking Accounts

### PR DESCRIPTION
**Pull request by fleko**

### CODE REVIEW

File: 05-Linking-Accounts/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+__Note:__ If you are using Windows, uncomment the `tzinfo-data` gem in the gemfile.
+
+## Important Snippets
+
+### 1. Auth0 Lock Account Linking Setup
+```js
+function linkPasswordAccount(){
+	connection = $("#link_provider").val();
+  var linkOptions = {
+		auth: {
```
Comment (falvarisqueta) [inline 05-Linking-Accounts/README.md:35]: indentation looks odd

File: 05-Linking-Accounts/app/assets/javascripts/home.js.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	 };
+	var lock = new Auth0Lock('<%= Rails.application.secrets.auth0_client_id %>', '<%= Rails.application.secrets.auth0_domain %>', options);
+
+ 	lock.show();
+}
+
+function linkPasswordAccount(){
+	connection = $("#link_provider").val();
+  var linkOptions = {
+		auth: {
```
Comment (falvarisqueta) [inline 05-Linking-Accounts/app/assets/javascripts/home.js.erb:18]: indentation looks odd

File: 05-Linking-Accounts/app/controllers/auth0_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    # OmniAuth places the User Profile information (retrieved by omniauth-auth0) in request.env['omniauth.auth'].
+    # In this tutorial, you will store that info in the session, under 'userinfo'.
+    # If the id_token is needed, you can get it from session[:userinfo]['credentials']['id_token'].
+    # Refer to https://github.com/auth0/omniauth-auth0#auth-hash for complete information on 'omniauth.auth' contents.
+    unless session[:userinfo].present?
+      session[:userinfo] = request.env['omniauth.auth']
+      redirect_to '/dashboard'
+      return
+    end
+    session[:linkuserinfo] = request.env['omniauth.auth']
```
Comment (falvarisqueta) [inline 05-Linking-Accounts/app/controllers/auth0_controller.rb:13]: looks like session stores same value for userinfo and linkuserinfo. Is this expected?

File: 05-Linking-Accounts/app/assets/javascripts/home.js.erb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    	title: 'Link another account'
+    }
+  };
+  if (connection){
+    linkOptions.allowedConnections = [connection];
+  }
+	var linkLock = new Auth0Lock('<%= Rails.application.secrets.auth0_client_id %>', '<%= Rails.application.secrets.auth0_domain %>', linkOptions);
+
+  //open lock in signin mode, with the customized options for linking
+  linkLock.show();
```
Comment (falvarisqueta) [inline 05-Linking-Accounts/app/assets/javascripts/home.js.erb]: looks like linking account happens on JS. Shouldn't we approach with a Ruby solution?

File: 05-Linking-Accounts/app/controllers/auth0_controller.rb (hunk 3 above)
Comment (fleko) [inline 05-Linking-Accounts/app/controllers/auth0_controller.rb:13]: @falvarisqueta: Yes, but it's not the same value. On the first login, it's set to userinfo (main identity) and on second login (account to be linked) it is set to linkuserinfo

File: 05-Linking-Accounts/app/assets/javascripts/home.js.erb (hunk 4 above)
Comment (fleko) [inline 05-Linking-Accounts/app/assets/javascripts/home.js.erb]: @falvarisqueta: not really, we need to set lock in such a way that it "forgets" the previous login, but the linkage occurs in the settings controller using the SDK (see line 97 of settings controller)

**Pull request merged**